packages/core/data/mm/world/overworld.yml

```diff
@@ -840,49 +896,51 @@
   region: DEKU_PALACE
   exits:
     "Swamp Back": "true"
-    "Deku Palace Cliff": "has(MASK_DEKU)"
+    "Deku Palace Cliff": "has(MASK_DEKU) || short_hook_anywhere"
     "Near Deku Shrine": "event(CLEAN_SWAMP)"
-    "Deku Palace Main": "has(MASK_DEKU) || trick(MM_PALACE_GUARD_SKIP)"
+    "Deku Palace Main": "has(MASK_DEKU) || trick(MM_PALACE_GUARD_SKIP) || short_hook_anywhere"
     "Deku Palace Upper": "(event(CLEAN_SWAMP) || has(MASK_DEKU)) && can_use_beans"
   events:
     PICTURE_BIG_OCTO: "soul_octorok && has(MASK_DEKU) && has(PICTOGRAPH_BOX)"
-    NUTS: "soul_deku_baba && (has(MASK_DEKU) || (event(CLEAN_SWAMP) && (can_fight || has_arrows || has_explosives || can_hookshot_short)))"
+    NUTS: "soul_deku_baba && (has(MASK_DEKU) || short_hook_anywhere || (event(CLEAN_SWAMP) && (can_fight || has_arrows || has_explosives || can_hookshot_short)))"
 "Deku Palace Main":
   region: DEKU_PALACE
   exits:
     "Deku Palace Throne": "true"
     "Deku Palace Front": "true"
     "Deku Palace Grotto": "true"
-    "Deku Palace Upper": "trick(MM_PALACE_BEAN_SKIP)"
+    "Deku Palace Upper": "trick(MM_PALACE_BEAN_SKIP) || short_hook_anywhere"
   locations:
     "Deku Palace HP": "true"
 "Deku Palace Upper":
   region: DEKU_PALACE
   events:
-    MAGIC: "has(MASK_DEKU)"
+    MAGIC: "has(MASK_DEKU) || short_hook_anywhere"
   exits:
     "Deku Palace Main": "true"
-    "Deku Palace Near Cage": "has(MASK_DEKU)"
+    "Deku Palace Near Cage": "has(MASK_DEKU) || short_hook_anywhere"
   locations:
     "Deku Palace Pot 1": "true"
     "Deku Palace Pot 2": "true"
 "Deku Palace Throne":
   region: DEKU_PALACE
   exits:
     "Deku Palace Main": "true"
+    "Deku Palace Cage": "event(RETURN_PRINCESS) && short_hook_anywhere"
   events:
     PICTURE_DEKU_KING: "has(PICTOGRAPH_BOX) && has(MASK_DEKU)"
     RETURN_PRINCESS: "has_deku_princess && has(MASK_DEKU)"
 "Deku Palace Near Cage":
   region: DEKU_PALACE
   exits:
     "Deku Palace Main": "true"
-    "Deku Palace Upper": "has(MASK_DEKU) && soul_deku_scrub"
+    "Deku Palace Upper": "has(MASK_DEKU) || short_hook_anywhere"
```

Comment:
reason of removing the scrub soul?

Reply:
you don't need to defeat any scrubs to get across.

Reply:
I was thinking the other way around